### ISSUE TITLE
smap and amap not listed in __all__

### ISSUE THREAD

**Issue by akhilman** [vxgmichel/aiostream#20]
[`smap`][1] and [`amap`][2] operators not accessible with `aiostream.stream`. Please add this operators to [`__all__`][3] in `aiostream.stream.combine`.

[1]: https://github.com/vxgmichel/aiostream/blob/de0bdabfe3a4c320300f4e85982b5277243581f2/aiostream/stream/combine.py#L56
[2]: https://github.com/vxgmichel/aiostream/blob/de0bdabfe3a4c320300f4e85982b5277243581f2/aiostream/stream/combine.py#L75
[3]: https://github.com/vxgmichel/aiostream/blob/de0bdabfe3a4c320300f4e85982b5277243581f2/aiostream/stream/combine.py#L12

**Comment by vxgmichel** [comment]
Hi @akhilman, thanks for creating this issue.

I'm not convinced those operators are worth being exposed though, since `map` already picks one or the other depending on the provided function. I can only think of two use cases where `map` is not enough:

**1.** A coroutine function is provided, but it's not meant to be awaited right away:

```python
async def _await(arg):
    return await arg

xs = stream.iterate([1, 2, 3])
ys = stream.combine.smap(xs, asyncio.sleep)
zs = stream.map(ys, _await)
await zs  # wait 6 seconds
```
This feels weird and unlikely.

**2.** The provided function is not a coroutine function but it's meant to be awaited anyway:

```python
xs = stream.iterate([1, 2, 3])
ys = stream.combine.amap(xs, lambda x: asyncio.sleep(x, x))
zs = stream.print(ys)
await zs  # wait 6 seconds and print 1 2 3
```

This seems more useful, although there are other ways to achieve the same purpose like defining the coroutine function explicitly:

```python
async def corofn(x):
    await asyncio.sleep(x, x)
[...]
ys = stream.map(xs, corofn)
```
Or use some kind of `_async` helper:
```python
def _async(fn):
    @functools.wraps(fn)
    async def wrapper(*args, **kwargs):
        return await fn(*args, **kwargs)
    return wrapper
[...]
ys = stream.map(xs, _async(lambda x: asyncio.sleep(x, x)))
```
What do you think?

**Comment by akhilman** [comment]
Greetings @vxgmichel . Thanks for explanation. Seems like in most cases `map()` works with coroutines as expected with only one exception where I use `amap()` with `lambda` expression.

My first idea was to build a new operator:
```python
@aiostream.operator(pipable=True)
async def stream_await(src):
    with aiostream.stream.iterate(src).stream() as streamer:
        async for ret in streamer:
            if asyncio.iscoroutine(ret):
                ret =  await ret
            yield ret
```
But then I found that `amap()` does exactly why I need.
`_async` wrapper looks ok for me.

**Comment by akhilman** [comment]
But exposing `amap()` is more obvious and less magical solution for me.

**Comment by vxgmichel** [comment]
I'm actually tempted to expose `_await` and `_async` in `aiostream`. This would allow the following code:
```python
from aiostream import stream, _async, _await
...
# _async approach
xs = ...
ys = stream.map(xs, _async(lambda x: ...))

# _await approach
xs = ...
ys = stream.map(xs, lambda x: ...)
zs = stream.map(ys, _await)
```

I think it's acceptable.

Another solution is to add a `force_await=False` flag to `map` but it has already [many arguments](http://aiostream.readthedocs.io/en/latest/operators.html#aiostream.stream.map).

**Comment by vxgmichel** [comment]
I don't think it's going to happen anytime soon ;)

**Comment by vxgmichel** [comment]
The `async_` and `await_` helpers are now available in the aiostream namespace (PR #21), they will become available with version 0.3.1. In the meantime, using `stream.combine.amap` is fine.

Thanks again :)
